src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/HeapHprofBinWriter.java

```diff
@@ -581,23 +581,24 @@ private int calculateArrayMaxLength(long originalArrayLength,
                                         String typeName) throws IOException {
 
         long length = originalArrayLength;
-
-        // now get the current position to calculate length
-        long dumpEnd = fos.getChannel().position();
         long originalLengthInBytes = originalArrayLength * typeSize;
+        long currentRecordLength = 0;
 
-        // calculate the length of heap data
-        // only process when segmented heap dump is not used, since SegmentedOutputStream
-        // could create segment automatically.
-        long currentRecordLength = (dumpEnd - currentSegmentStart - 4L);
```

Comment:
As you moved this initialization inside the `if (!useSegmentedHeapDump)` condition then the `currentRecordLength` will be now equal to 0 for segmented heap dump as well.
Could you, please, confirm it was your intention?